composer.json

```diff
@@ -7,6 +7,7 @@
     "license": "GPL-2.0+",
     "require": {
         "php": "~7.0",
+        "joomla/event": "~2.0",
```

Comment:
Surely the point of catching the `UnexpectedValueException` in `dispatchEvent` is that this isn't a requirement?

Reply:
To make the classes dispatcher aware it is.

Reply:
OK So it's weird. `getDispatcher` is only in the trait, not the interface, so even if we pretend the interface doesn't exist we end up in massive problems. But at the same time, at the point we are enforcing the event package, for 2.0, is it worth catching exceptions if the event package isn't present? surely we just require it outright?

Reply:
It catches if the dispatcher isn't set based on the trait's internals.  It's  not catching if the package isn't present.

If the package weren't present then PHP would fatal out because it wouldn't be able to compose the class (interface/trait missing).  The only way to make the package an optional dependency is to not use the interface and trait.